src/Handlers/SystemWeb/HttpHandlerEndpointConventionBuilder.cs

```diff
@@ -78,7 +78,11 @@ private IEnumerable<IHttpHandlerMetadata> CollectMetadata()
 
         foreach (var mappedRoute in mappedRoutes)
         {
-            metadataCollection.Add(mappedRoute.Route, new MappedHandlerMetadata(mappedRoute.Route, metadataCollection[mappedRoute.Path]));
+            // TODO should we log if we can't find it? It may be a race condition where the compilation hasn't found it yet, so it could be an unnecessary warning
```

Comment:
if it's unnessary, we should ignore it...

Reply:
It may be nice to notify a warning or something if a route collection is registered but the route isn't found. Not sure how webforms on framework handled it, so I'll create an issue to investigate.